android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java

```diff
@@ -315,35 +314,41 @@ public void setNetworkWithAccountCheck(
             NetworkInfo networkToBeSetAsSelected, Callbacks.Callback1<Boolean> callback) {
         NetworkInfo selectedNetwork = _mDefaultNetwork.getValue();
         if (isSameNetwork(networkToBeSetAsSelected, selectedNetwork)) return;
-        boolean hasAccountOfNetworkType = hasAccountOfNetworkType(networkToBeSetAsSelected);
-        if (hasAccountOfNetworkType) {
-            mJsonRpcService.setNetwork(networkToBeSetAsSelected.chainId,
-                    networkToBeSetAsSelected.coin, mOrigin, isSelected -> {
-                        callback.call(isSelected);
-                        mCryptoActions.updateCoinType();
-                        init();
-                    });
-        } else {
-            _mNeedToCreateAccountForNetwork.postValue(networkToBeSetAsSelected);
-            callback.call(false);
-        }
+
+        mBraveWalletService.ensureSelectedAccountForChain(
+                networkToBeSetAsSelected.coin, networkToBeSetAsSelected.chainId, accountId -> {
+                    if (accountId == null) {
+                        _mNeedToCreateAccountForNetwork.postValue(networkToBeSetAsSelected);
+                        callback.call(false);
+                        return;
+                    }
+                    setNetworkForSelectedAccountOnActiveOrigin(networkToBeSetAsSelected, callback);
+                });
     }
 
     public void setNetworkWithAccountCheck(String chainId, Callbacks.Callback1<Boolean> callback) {
         NetworkInfo networkToBeSetAsSelected = getNetwork(chainId);
         setNetworkWithAccountCheck(networkToBeSetAsSelected, callback);
     }
 
-    public void setNetwork(
+    public void setNetworkForSelectedAccountOnActiveOrigin(
             NetworkInfo networkToBeSetAsSelected, Callbacks.Callback1<Boolean> callback) {
-        mJsonRpcService.setNetwork(networkToBeSetAsSelected.chainId, networkToBeSetAsSelected.coin,
-                mOrigin, isSelected -> {
-                    callback.call(isSelected);
+        mBraveWalletService.setNetworkForSelectedAccountOnActiveOrigin(
+                networkToBeSetAsSelected.chainId, success -> {
+                    callback.call(success);
```

Comment:
Use `mOrigin` and don't use activeOrign calls for android as it can be from any active tab.

Reply:
Made it so default/activeOrigin network setting is picked depending on panel/wallet ui

Reply:
so now `GetActiveOrigin` will automatically pick the correct domain based on tab or wallet? if yes then do we need to revert this https://github.com/brave/brave-core/pull/18930#discussion_r1252799154?

Reply:
mOrigin should work for `_mChainId.addSource(...)`. I'm not going to refactor it to use `setNetworkForSelectedAccountOnActiveOrigin` and `setDefaultNetwork` as it not easy to implement correctly in small set of changes.
Technically that should be two _mChainId live variables - one for panel network which depends on account and origin, another for default(wallet) network which depends only on account.
Will do that in future PR

Reply:
Not sure if we need or want to maintain states for panel and wallet separately though just prefers to use one path, either activeOrigin... calls etc. (as you mentioned, now these calls works automatically for wallet and panel) or use `mOrigin`.
Anyway, Just test these changes for network selection for dapps (per origin) and wallet.

Reply:
This confusion comes from desktop ui which doesn't have 'native' wallet ui and always have associated origin(modulo corner cases like opaque origin) which is either an origin like https://dapp.site.com(panel per origin networks) or https://brave/wallet(wallet default network).

But actually these are two different entities.

I did some testing and will do another try before merging after all reviewers are ok